modules/hdf/test/test_hdf5.cpp

```diff
@@ -15,7 +17,7 @@ struct HDF5_Test : public testing::Test
 {
     virtual void SetUp()
     {
-        m_filename = "test.h5";
+        m_filename = std::tmpnam(nullptr) + "_test.h5";
```

Comment:
It's recommended to use `cv::tempfile` for tests: https://docs.opencv.org/4.x/db/de0/group__core__utils.html#ga9fcedbe6d0104d7ce012987478b060d4